test/exe/main_common_test.cc

```diff
@@ -7,71 +7,120 @@
 // TODO(issues/2649): This test needs to be parameterized on IP versions.
 #ifndef ENVOY_CONFIG_COVERAGE
 
+#include <unistd.h>
+
+#include "common/runtime/runtime_impl.h"
+
 #include "exe/main_common.h"
 
 #include "server/options_impl.h"
 
 #include "test/test_common/environment.h"
+#include "test/test_common/utility.h"
 
 #include "gtest/gtest.h"
 
 #ifdef ENVOY_HANDLE_SIGNALS
 #include "exe/signal_action.h"
 #endif
 
-#ifdef ENVOY_HOT_RESTART
-#include "server/hot_restart_impl.h"
-#endif
-
 namespace Envoy {
 
-TEST(MainCommon, ConstructDestruct) {
+/**
+ * Captures common functions needed for invoking MainCommon. Generates a
+ * unique --base-id setting based on the pid and a random number. Maintains
+ * an argv array that is terminated with nullptr. Identifies the config
+ * file relative to $TEST_RUNDIR.
+ *
+ * TODO(jmarantz): Make these tests work with ipv6. See
+ * https://github.com/envoyproxy/envoy/issues/2649
+ */
+class MainCommonTest : public testing::Test {
+public:
+  MainCommonTest()
+      : config_file_(Envoy::TestEnvironment::getCheckedEnvVar("TEST_RUNDIR") +
+                     "/test/config/integration/google_com_proxy_port_0.v2.yaml"),
+        random_string_(fmt::format("{}", computeBaseId())),
+        argv_({"envoy-static", "--base-id", random_string_.c_str(), nullptr}) {}
+
+  /**
+   * Computes a numeric ID to incorporate into the names of shared-memory segments and
+   * domain sockets, to help keep them distinct from other tests that might be running
+   * concurrently.
+   *
+   * @return uint32_t a unique numreic ID based on PID and a random number.
+   */
+  static uint32_t computeBaseId() {
+    Runtime::RandomGeneratorImpl random_generator_;
+    // Pick a prime number to give more of the 32-bits of entropy to the PID, and the
+    // remainder to the random number.
+    const uint32_t four_digit_prime = 7919;
+    return getpid() * four_digit_prime + random_generator_.random() % four_digit_prime;
```

Comment:
This seems overkill; i would either just use the PID or random(), I don't think we're benefiting that much from combining.